rapidsai/cudf
# [REVIEW] Enable different RMM allocation modes in unit tests

Test infrastructure changes:

- Add test environment to initialize RMM, instead of initializing it once per test suite.
- Use a custom main function instead of gtest one to allow parsing the rmm command line parameter and instantiating the test environment.
- RMM pool allocation mode can be set via command line parameters.
- The pool mode is used by default.

Other fixes (mainly for issues exposed by the pool allocator):

- When comparing null masks, only compare the relevant bits, not padding.
- Use legacy fixture for all legacy tests.
- Replaces erroneous uses of expect_column_properties_equal with expect_columns_equal.
- Remove null count check from expect_column_properties_equal (not a column property).
- ~~Fix a dangling pointer error in create_test_string().~~ (obsolete after latest merge)
- Add helper check_float_column() to JSON tests to make up for the change in expect_column_properties_equal behavior.

Closes #797
